addons/website_mail/controllers/main.py

```diff
@@ -47,111 +46,90 @@ def _message_post_helper(res_model='', res_id=None, message='', token='', token_
-                partner_ids = [partner_obj.create(cr, SUPERUSER_ID, {'name': name, 'email': email}, context=context)]
-
+                partner_ids = request.env['res.partner'].sudo().create({'name': name, 'email': email}).ids
         # add or remove follower
-        if _message_is_follower:
-            _object.check_access_rule(cr, uid, [_id], 'read', context)
-            _object.message_unsubscribe(cr, SUPERUSER_ID, [_id], partner_ids, context=context)
+        if is_follower:
+            record.check_access_rule('read')
+            record.sudo().message_unsubscribe(partner_ids)
             return False
         else:
-            _object.check_access_rule(cr, uid, [_id], 'read', context)
+            record.check_access_rule('read')
             # add partner to session
             request.session['partner_id'] = partner_ids[0]
-            _object.message_subscribe(cr, SUPERUSER_ID, [_id], partner_ids, context=context)
+            record.sudo().message_subscribe(partner_ids)
             return True
 
     @http.route(['/website_mail/is_follower'], type='json', auth="public", website=True)
     def call(self, model, id, **post):
-        id = int(id)
-        cr, uid, context = request.cr, request.uid, request.context
-
-        partner_obj = request.registry.get('res.partner')
-        users_obj = request.registry.get('res.users')
-        obj = request.registry.get(model)
-
-        partner_id = None
-        public_id = request.website.user_id.id
-        if uid != public_id:
-            partner_id = users_obj.browse(cr, SUPERUSER_ID, uid, context).partner_id
+        user = request.env.user
+        partner = None
+        public_user = request.website.user_id
+        if user != public_user:
+            partner = request.env.user.partner_id
         elif request.session.get('partner_id'):
-            partner_id = partner_obj.browse(cr, SUPERUSER_ID, request.session.get('partner_id'), context)
-        email = partner_id and partner_id.email or ""
+            partner = request.env['res.partner'].sudo().browse(request.session.get('partner_id'))
 
         values = {
-            'is_user': uid != public_id,
-            'email': email,
+            'is_user': user != public_user,
+            'email': partner.email if partner else "",
             'is_follower': False,
             'alias_name': False,
         }
 
-        if not obj:
-            return values
-        obj_ids = obj.exists(cr, SUPERUSER_ID, [id], context=context)
-        if obj_ids:
-            if partner_id:
-                values['is_follower'] = len(
-                    request.registry['mail.followers'].search(
-                        cr, SUPERUSER_ID, [
-                            ('res_model', '=', model),
-                            ('res_id', '=', obj_ids[0]),
-                            ('partner_id', '=', partner_id.id)
-                        ], context=context)) == 1
+        record = request.env[model].sudo().browse(int(id)).exists()
```

Comment:
Actually I don't think this line is necessary. Is the record does not exist the entries in mail_followers table will be deleted.